classes/notification/managerDelegate/EditingStatusNotificationManager.inc.php

```diff
@@ -0,0 +1,218 @@
+ *
+ * Copyright (c) 2014-2016 Simon Fraser University Library
+ * Copyright (c) 2003-2016 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class EditingStatusNotificationManager
+ * @ingroup classses_notification_managerDelegate
+ *
+ * @brief Editing status notifications types manager delegate.
+ */
+
+import('lib.pkp.classes.notification.NotificationManagerDelegate');
+
+class EditingStatusNotificationManager extends NotificationManagerDelegate {
+
+	/**
+	 * Constructor.
+	 * @param $notificationType int NOTIFICATION_TYPE_...
+	 */
+	function EditingStatusNotificationManager($notificationType) {
+		parent::NotificationManagerDelegate($notificationType);
+	}
+
+	/**
+	 * @copydoc PKPNotificationOperationManager::getNotificationMessage()
+	 */
+	public function getNotificationMessage($request, $notification) {
+		switch ($notification->getType()) {
+			case NOTIFICATION_TYPE_ASSIGN_COPYEDITOR:
+				return __('notification.type.assignCopyeditors');
+			case NOTIFICATION_TYPE_AWAITING_COPYEDITS:
+				return __('notification.type.awaitingCopyedits');
+		}
+	}
+
+	/**
+	 * @copydoc PKPNotificationOperationManager::getNotificationUrl()
+	 */
+	public function getNotificationUrl($request, $notification) {
+		$url = parent::getNotificationUrl($request, $notification);
+		$dispatcher = Application::getDispatcher();
+		$contextDao = Application::getContextDAO();
+		$context = $contextDao->getById($notification->getContextId());
+
+		switch ($notification->getType()) {
+			case NOTIFICATION_TYPE_ASSIGN_COPYEDITOR:
+			case NOTIFICATION_TYPE_AWAITING_COPYEDITS:
+				assert($notification->getAssocType() == ASSOC_TYPE_SUBMISSION && is_numeric($notification->getAssocId()));
+				return $dispatcher->url($request, ROUTE_PAGE, $context->getPath(), 'workflow', 'access', $notification->getAssocId());
+		}
+	}
+
+	/**
+	 * @copydoc PKPNotificationOperationManager::getStyleClass()
+	 */
+	public function getStyleClass($notification) {
+		return NOTIFICATION_STYLE_CLASS_INFORMATION;
+	}
+
+	/**
+	 * @copydoc NotificationManagerDelegate::updateNotification()
+	 */
+	public function updateNotification($request, $userIds, $assocType, $assocId) {
+		$context = $request->getContext();
+		$contextId = $context->getId();
+
+		assert($assocType == ASSOC_TYPE_SUBMISSION);
+		$submissionId = $assocId;
+		$submissionDao = Application::getSubmissionDAO();
+		$submission = $submissionDao->getById($submissionId);
+
+		$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO');
+		$editorStageAssignments = $stageAssignmentDao->getEditorsAssignedToStage($submissionId, $submission->getStageId());
+
+		// One way to check if a copyeditor is assigned
+		// Get the 'copyedit assignment' notifications
+		$copyEditorAssigned = false;
+		$notificationDao = DAORegistry::getDAO('NotificationDAO'); /* @var $notificationDao NotificationDAO */
+		$copyEditAssignmentNotificationFactory = $notificationDao->getByAssoc(
```

Comment:
No queries are automatically created. I will however put everything in one class, I think this will work well. I could also leave the notification check for now and remove it later, when we realized what to do with and implemented automatic queries. Just let me know what do you think would be best for now: with or without notification check?